### ISSUE TITLE
system V doesn't seem to handle MySQL root password

### ISSUE THREAD

**Issue by krodelabestiole** [Microsoft/WSL#3293]
Long story short, I can't handle MySQL service normally when using a password for MySQL root account :
````
kro@minipoutre ~  sudo service mysql start
[ ok ] Starting MariaDB database server: mysqld ..
kro@minipoutre ~  ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: NO)
^C
kro@minipoutre ~  sudo service mysql status
/usr/bin/mysqladmin: connect to server at 'localhost' failed
error: 'Access denied for user 'root'@'localhost' (using password: NO)'
[info] .
kro@minipoutre ~  sudo service mysql stop
[FAIL] Stopping MariaDB database server: mysqld failed!
kro@minipoutre ~  sudo killall mysqld
kro@minipoutre ~  sudo service mysql status
[info] MariaDB is stopped..
````

It works fine only with no password.
I have set the password with `mysql_secure_installation`.

Appart from this MySQL (MariaDB here) seems to work fine.
I have the problem with any distro I tried : Stretch, Xenial, Bionic

Any idea what is causing the issue and how to fix it ?

**Comment by therealkenc** [comment]
This one was chirping crickets because it isn't clear what variation of mysql you are running, and the issue template was deleted. Please supply your CLI repro steps leading up to `sudo service mysql start` as well as your Windows version. Linking a gist of the `strace` log might also be helpful, although I am not sure this is going to be a syscall thing.

**Comment by glachancecmaisonneuve** [comment]
Tried to repro.  I think the issue may be that before mysql is properly configured, pretty much any command, including but not limited to service start mysql will return "bad username or password".  After running ```mysql_secure_installation``` (and now having a decent config), the error messages start getting more diverse.

TLDR: probably a configuration mysql error.  should close.

**Comment by krodelabestiole** [comment]
Sorry I was performing different tests to document the use of MySQL in a french community.
I can't remember what this exact problem was but it was most likely due to the use of auth_socket by default on ubuntu bionic I was not expecting.
mysql_secure_installation has became totally useless on debian / ubuntu based system from what I came to understand.
I should have closed this issue myself but I was doing 10 things at the same time, sorry, and thank you.
